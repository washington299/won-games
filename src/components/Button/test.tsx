import { screen } from "@testing-library/react";

import { renderWithTheme } from "utils/tests/helpers";

import { Button } from ".";

describe("<Button />", () => {
	it("should render button text correctly", () => {
		renderWithTheme(<Button>My button</Button>);

		expect(screen.getByText(/My button/i)).toBeInTheDocument();
	});

	it("should render button with medium size by default", () => {
		renderWithTheme(<Button>My button</Button>);

		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ height: "4rem" });
		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ fontSize: "1.4rem" });
	});

	it("should render button with small size when prop is passed", () => {
		renderWithTheme(<Button size="small">My button</Button>);

		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ height: "3rem" });
		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ fontSize: "1.2rem" });
	});

	it("should render button with large size when prop is passed", () => {
		renderWithTheme(<Button size="large">My button</Button>);

		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ height: "5rem" });
		expect(screen.getByRole("button", { name: /My button/i })).toHaveStyle({ fontSize: "1.6rem" });
	});
});
