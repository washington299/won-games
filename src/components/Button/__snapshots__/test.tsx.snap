// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button text correctly 1`] = `
.c0 {
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  background: linear-gradient(180deg,#ff5f5f -15%,#f062c0 100%,#f23131 100%);
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  cursor: pointer;
}

.c0 svg {
  width: 2rem;
}

<button
  class="c0"
>
  <span>
    My button
  </span>
</button>
`;
